src/sources/kubernetes/file_source_builder.rs

```diff
@@ -0,0 +1,222 @@
+
+        let (file_send, file_recv) = mpsc::channel(1000);
+        let file_source = self
+            .file_config
+            .build("file_source", globals, file_send)
+            .map_err(|e| format!("Failed in creating file source with error: {:?}", e))?;
+
+        Ok((file_recv, file_source))
+    }
+
+    /// Configures include in FileConfig
+    fn file_source_include(&mut self) -> crate::Result<()> {
+        // Prepare include patterns
+
+        // Contains patterns that match on all possible included UIDs.
+        let include_pod_uids = self
+            .numerical_pod_uids()?
+            .into_iter()
+            .flat_map(|uid| to_uid_forms(uid.as_str()))
+            .collect();
+
+        // Pattern that matches to all UIDs, and only UIDs.
+        let any_uid = to_uid_forms("");
+        let pod_uids = not_empty_or_else(&include_pod_uids, &any_uid);
+
+        // Contains patterns: container_name*
+        let include_container_names = self
+            .config
+            .include_container_names
+            .iter()
+            .map(|name| name.clone() + "*")
+            .collect();
+
+        let any_name = vec!["*".to_string()];
+        let container_names = not_empty_or_else(&include_container_names, &any_name);
+
+        let any_namespace = vec!["*".to_string()];
+        // Will match only on exactly included namespaces, or any if none specified.
+        let namespaces = not_empty_or_else(&self.config.include_namespaces, &any_namespace);
+
+        // The following creates and adds path includes of form:
+        // LOG_DIRECTORY/uid/container_name/*.log
+        //
+        // To construct the paths, it's necessary to make every
+        // possible combination of uid and container_name.
+        for uid in pod_uids {
+            for container_name in container_names {
+                self.file_config.include.push(
+                    (LOG_DIRECTORY.to_owned()
+                        + format!("{}/{}/*.log", uid, container_name).as_str())
+                    .into(),
+                );
+            }
+        }
+
+        // The following creates and adds path includes of form:
+        // LOG_DIRECTORY/namespace_*_uid/container_name/*.log
+        //
+        // To construct the paths, it's necessary to make every
+        // possible combination of uid, container_name, and namespace.
+        for uid in pod_uids {
+            for container_name in container_names {
+                for namespace in namespaces {
+                    self.file_config.include.push(
+                        (LOG_DIRECTORY.to_owned()
+                            + format!("{}_*_{}/{}/*.log", namespace, uid, container_name).as_str())
+                        .into(),
+                    );
+                }
+            }
+        }
+
+        Ok(())
+    }
+
+    /// Configures exclude in FileConfig
+    fn file_source_exclude(&mut self) {
```

Comment:
I would add a comment here saying that currently, this provides a exclude if no other settings were set. We should ensure that we document this default as this is an interesting behavior affected by multiple config values.